backend/routes.js

```diff
@@ -447,6 +447,99 @@ route.POST('/deleteFile/{hash}', {
   return h.response()
 })
 
+route.POST('/kv/{contractID}/{key}', {
+  auth: {
+    strategies: ['chel-shelter'],
+    mode: 'required'
+  },
+  payload: {
+    parse: false,
+    maxBytes: 6 * MEGABYTE, // TODO: make this a configurable setting
+    timeout: 10 * SECOND // TODO: make this a configurable setting
+  }
+}, async function (request, h) {
+  const { contractID, key } = request.params
+  console.debug(`POST /kv/${contractID}/${key}`)
+
+  if (key.startsWith('_private')) {
+    return Boom.notFound()
+  }
+
+  if (!ctEq(request.auth.credentials.billableContractID, contractID)) {
+    return Boom.unauthorized(null, 'shelter')
+  }
+
+  const existing = await sbp('chelonia/db/get', `_private_kv_${contractID}_${key}`)
+
+  // Some protection against accidental overwriting by implementing the if-match
+  // header
```

Comment:
A bit more explanation here would be helpful, e.g. the specific scenario that this logic helps with